### PR TITLE
Add Liberapay button in the donate section

### PR DESCRIPTION
**Pull request by Flaburgan**
https://liberapay.com/

Liberapay is a recurrent donations platform (quite like patreon, but freesoftware ;)). I'd like to integrate its donation button (without any external requests :)) to the donate section of the left side bar.

### CODE REVIEW

File: app/views/streams/main_stream.html.haml [hunk 1]
```diff
@@ -123,8 +123,7 @@
               = t("bookmarklet.heading")
           .content
             != t("bookmarklet.explanation", link: link_to(t("bookmarklet.post_something"), bookmarklet_code))
-
-        - if AppConfig.settings.paypal_donations.enable? || AppConfig.bitcoin_donation_address
+        - if AppConfig.settings.paypal_donations.enable? || AppConfig.settings.liberapay.enable? || AppConfig.bitcoin_donation_address
```
Comment (diaspora-code-review) [inline app/views/streams/main_stream.html.haml]: Line is too long. [134/120]

Comment (SuperTux88) [inline app/views/streams/main_stream.html.haml]: We should create a method `donations_enabled?` in the `AppConfig` or `ApplicationHelper`.

File: app/views/shared/_donatepod.html.haml [hunk 2]
```diff
@@ -9,7 +9,11 @@
     %input{name: "currency_code", type: "hidden", value: AppConfig.settings.paypal_donations.currency}
     %input{name: "submit", type: "submit", value: t("aspects.index.donate"), class: "btn btn-default"}
 
+- if AppConfig.settings.liberapay.enable?
+  Liberapay:
+  %a{href: "https://liberapay.com/#{AppConfig.settings.liberapay.username}/donate"}
+    = image_tag "buttons/liberapay-button.svg", alt: "Donate", title: "Donate to liberapay"
```
Comment (SuperTux88) [inline app/views/shared/_donatepod.html.haml]: Please make these Strings translatable.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -23,6 +23,10 @@ def source_url
     AppConfig.settings.source_url.presence || "#{root_path.chomp('/')}/source.tar.gz"
   end
 
+  def donations_enabled?
+    AppConfig.settings.paypal_donations.enable? || AppConfig.settings.liberapay.username.present? || AppConfig.bitcoin_donation_address
```
Comment (diaspora-code-review) [inline app/helpers/application_helper.rb]: Line is too long. [135/120]

File: features/step_definitions/custom_web_steps.rb [hunk 4]
```diff
@@ -288,6 +288,14 @@ def take_screenshots_with_login
   find("#bitcoin_address")['value'].should == "AAAAAA"
 end
 
+Given /^I have configured a Liberapay username$/ do
+  AppConfig.settings.liberapay.username = "BBBBBB"
+end
+
+Then /^I should see the Liberapay donate button$/ do
+  find("#liberapay-button")['href'].should == "https://liberapay.com/BBBBBB/donate"
```
Comment (diaspora-code-review) [inline features/step_definitions/custom_web_steps.rb]: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.

**Comment by Flaburgan** [comment]
Corrected the remarks and added a test ;)

File: app/views/shared/_donatepod.html.haml [hunk 5]
```diff
@@ -9,7 +9,11 @@
     %input{name: "currency_code", type: "hidden", value: AppConfig.settings.paypal_donations.currency}
     %input{name: "submit", type: "submit", value: t("aspects.index.donate"), class: "btn btn-default"}
 
+- if AppConfig.settings.liberapay.username.present?
+  Liberapay:
+  %a{id: "liberapay-button", href: "https://liberapay.com/#{AppConfig.settings.liberapay.username}/donate"}
+    = image_tag "buttons/liberapay-button.svg", alt: t("aspects.index.donate"), title: t("aspects.index.donate_liberapay")
```
Comment (diaspora-code-review) [inline app/views/shared/_donatepod.html.haml]: Line is too long. [122/120]

Comment (SuperTux88) [inline app/views/shared/_donatepod.html.haml]: You can write the `title` in the next line.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]

**Comment by Flaburgan** [comment]
We should be ready to go, I even added the changelog entry ;)

File: app/views/streams/main_stream.html.haml [hunk 6]
```diff
@@ -123,8 +123,7 @@
               = t("bookmarklet.heading")
           .content
             != t("bookmarklet.explanation", link: link_to(t("bookmarklet.post_something"), bookmarklet_code))
-
```
Comment (SuperTux88) [inline app/views/streams/main_stream.html.haml]: Nit: lets keep this newline (all `section`s are separated with an empty line)

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
Sorry, I only looked at the code before, but now after testing it in the browser I saw that it's broken when you enable all 3 of the donation settings.

spectacle a14304

I think we should make it like this:

Paypal:  
[Button]  
Liberapay:  
[Button]  
Bitcoin:  
[input-field]

(currently the input field for bitcoin is also on the same line as "Bitcoin:", so it's theoretically out of scope here, but I would change that too.)

File: app/views/shared/_donatepod.html.haml [hunk 7]
```diff
@@ -1,15 +1,27 @@
...
-      %input{name: "encrypted", type: "hidden", value: AppConfig.settings.paypal_donations.paypal_unhosted_button_encrypted}
-    %input{name: "currency_code", type: "hidden", value: AppConfig.settings.paypal_donations.currency}
-    %input{name: "submit", type: "submit", value: t("aspects.index.donate"), class: "btn btn-default"}
+  %p
+    PayPal:
+    %br
+    %form{action: "https://www.paypal.com/cgi-bin/webscr", method: "post", target: "_blank"}
+      %input{name: "cmd", type: "hidden", value: "_s-xclick"}
+      - if AppConfig.settings.paypal_donations.paypal_hosted_button_id.present?
+        %input{name: "hosted_button_id", type: "hidden", value: AppConfig.settings.paypal_donations.paypal_hosted_button_id}
```
Comment (diaspora-code-review) [inline app/views/shared/_donatepod.html.haml]: Line is too long. [124/120]

File: app/views/shared/_donatepod.html.haml [hunk 8]
```diff
@@ -1,15 +1,27 @@
...
+      -if AppConfig.settings.paypal_donations.paypal_unhosted_button_encrypted.present?
```
Comment (diaspora-code-review) [inline app/views/shared/_donatepod.html.haml]: The - symbol should have one space separating it from code

File: app/views/shared/_donatepod.html.haml [hunk 9]
```diff
@@ -1,15 +1,27 @@
...
+        %input{name: "encrypted", type: "hidden", value: AppConfig.settings.paypal_donations.paypal_unhosted_button_encrypted}
```
Comment (diaspora-code-review) [inline app/views/shared/_donatepod.html.haml]: Line is too long. [126/120]

File: app/views/shared/_donatepod.html.haml [hunk 10]
```diff
@@ -1,15 +1,27 @@
...
+      %input{name: "currency_code", type: "hidden", value: AppConfig.settings.paypal_donations.currency}
+      %input{name: "submit", type: "submit", value: t("aspects.index.donate"), class: "btn btn-default"}
```
Comment (diaspora-code-review) [inline app/views/shared/_donatepod.html.haml]: Avoid defining `class` in attributes hash for static class names

File: app/views/shared/_donatepod.html.haml (hunk 7 above)
Comment (SuperTux88) [inline app/views/shared/_donatepod.html.haml]: Just write the `value` on the next line:
```haml
        %input{name: "hosted_button_id", type: "hidden",
               value: AppConfig.settings.paypal_donations.paypal_hosted_button_id}
```

File: app/views/shared/_donatepod.html.haml (hunk 9 above)
Comment (SuperTux88) [inline app/views/shared/_donatepod.html.haml]: Some as above, `value` on next line.

File: app/views/shared/_donatepod.html.haml (hunk 10 above)
Comment (SuperTux88) [inline app/views/shared/_donatepod.html.haml]:
```haml
%input.btn.btn-default{name: "submit", type: "submit", value: t("aspects.index.donate")}
```

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
Just fix the 3 last code-review remarks (I don't know why github marked it as outdated), and it should be good to merge :)

**Comment by Flaburgan** [comment]
I had already fixed one ;)
Everything should be fine now.

**Review by SuperTux88 — APPROVED** [review]
Thanks 👍 Pronto is happy now, and I'm happy too. Waiting for travis to be happy too... ;)

**Comment by SuperTux88** [comment]
Merged as 1028632ac1e393504516deaf588d4fbb7da1027d

Thank you :)

**Comment by trebmuh** [comment]
:+1: awesome

**Pull request closed without merging**